src/MacVim/MMVimController.h

```diff
@@ -28,6 +28,12 @@
     // TODO: Move all toolbar code to window controller?
     NSToolbar           *toolbar;
     NSMutableDictionary *toolbarItemDict;
+#if MAC_OS_X_VERSION_MAX_ALLOWED > MAC_OS_X_VERSION_10_12
```

Comment:
I could do that, but now I'm just wondering whether we even should check for this, or just remove the macro usage. This check only helps people *build* (not use) MacVim on older platforms but I presume most people who want to contribute to or build MacVim would want to have recent versions of macOS and Xcode. Are we going to keep supporting older versions of macOS development tools? I did notice the fullscreen code still checks for `MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_7` which seems crazy to me. (Most other code checks for MIN_ALLOWED, not MAX_ALLOWED)

Another important thing to check is **runtime** checking to make sure the program won't crash on users with older devices. I think the current implementation actually doesn't check that… `addMenuItemWithDescriptor` will call `addTouchbarItemWithLabel` which will start initializing `NSCustomTouchBarItem` which I think will crash in pre-Sierra machines (I don't have one to check and I'm not super familiar with Cocoa / Objective-C). I think we just need to inject some `NSClassFromString(@"UIAlertController")` checks before we start initializing the touch bar items, or delay making the touch bar items until we are in `makeTouchBar` which will only get called in systems supporting the touch bar as a delegate.

Reply:
> I could do that, but now I'm just wondering whether we even should check for this, or just remove the macro usage.

I'd like to, but I don't think we can. There are users out there who build MacVim from source on older versions of the OS, for a variety of reasons (such as not really liking some of Apple's changes in later versions, et cetera). MacVim has long supported those users, much like vim itself. 

Similar arguments pro and con can be made towards MacVim's support of legacy features (the old text renderer, the custom fullscreen mode) as well: these legacy features increase MacVim's maintenance cost, but there are users out there who really love them.

I don't think it is feasible to drop that support, as painful as it is to work with right now. 

However, it would be great to come up with an alternative. Branches were floated as an idea a while back, but it didn't really go anywhere.

Reply:
That's fair. I just added the better compile + runtime checks and pushed a change.